signer/src/transaction_coordinator.rs

```diff
@@ -164,7 +168,7 @@ where
                     // signer indicating that it has handled new requests.
                     Ok(SignerSignal::Event(SignerEvent::TxSigner(TxSignerEvent::NewRequestsHandled))) => {
                         tracing::debug!("received block observer notification");
-                        let _ = self.process_new_blocks().await
+                        let _ = self.process_new_blocks(delay_to_process_new_blocks).await
```

Comment:
At first I was going to write that it would be easier to have the delay here, but I actually like that putting it in `process_blocks()` _after_ the "are we the coordinator" check lets us skip the delay if we determine that we're not the coordinator.